--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -72,11 +72,13 @@ add_subdirectory(src/config)
 add_subdirectory(src/model)
 add_subdirectory(src/capture)
 add_subdirectory(src/reconstruction)
 add_subdirectory(src/streaming)
+add_subdirectory(src/streaming/webrtc)

 target_link_libraries(EdgeVision PRIVATE config)
 target_link_libraries(EdgeVision PRIVATE capture)
 target_link_libraries(EdgeVision PRIVATE model)
 target_link_libraries(EdgeVision PRIVATE reconstruction)
 target_link_libraries(EdgeVision PRIVATE streaming)
+target_link_libraries(EdgeVision PRIVATE streaming_webrtc)

 # ---- Bundle ----
